src/components/TimeCardPage/TimeSheet.js

```diff
@@ -49,33 +47,91 @@ export default function Page() {
         // Callback for date-range picker - does any pre-processing when grabbing a new date 
         setStartDate(startDate); 
         setEndDate(endDate); 
+        currentTimesheetsToDisplay(userTimesheets, start); 
         //TODO - remove once we are finished setting up API calls for  this 
         console.log("New date range has been selceted:\n\t %s \nto \n\t%s", start, end); 
     }
     
-    const [userTimesheets,setTimesheets] = useState([]); 
+    const [userTimesheets, setTimesheets] = useState([]); 
+    const [currentTimesheets, setCurrentTimesheets] = useState([]);
     const [selectedTimesheet, setTimesheet] = useState(); 
     const columns = defaultColumns 
 
     //Pulls user timesheets, marking first returned as the active one
     useEffect(() => {
         // Uncomment this if you want the default one loaded 
-        // setTimesheet(testingTimesheetResp)
+        //setTimesheet(testingTimesheetResp)
         apiClient.getUserTimesheets().then(timesheets => {
             setTimesheets(timesheets); 
-            //By Default just render / select the first timesheet for now  
-            if (timesheets.length > 0) {
-                setTimesheet(timesheets[0])
-            }  
-        });  
+            //By Default just render / select the first timesheet for now 
+            currentTimesheetsToDisplay(timesheets, startDate); 
+        });
     }, [])
 
     const processTimesheetChange = (timesheet) => {
         //Adding the time entry to the table 
         apiClient.addTimeEntry(timesheet); 
     }
 
+    const currentTimesheetsToDisplay = (timesheets, currentStartDate) => {
```

Comment:
⛏️ I think this would make more sense if it was named `setCurrentTimesheetsToDisplay`

Reply:
changed to setCurrentTimesheetToDisplay